pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -124,9 +134,14 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if vm != nil {
 			labels = &vm.Labels
 		}
-
-		nsxSubnet, err := r.SubnetService.GetSubnetByPath(nsxSubnetPath)
+		nsxSubnet, err := r.SubnetService.GetSubnetByPath(nsxSubnetPath, servicecommon.IsSharedSubnet(subnetCR))
```

Comment:
I do not understand what the below code on line #124 is for? Could you  help explain?
if !isExisting {
			defer r.SubnetPortService.ReleasePortInSubnet(nsxSubnetPath)
		}

Reply:
We have a map in SubnetPort service to track the number of SubnetPorts under creation on each Subnet.
!isExisting means the SubnetPort is newly created so we have already added one to the portcount for that subnet, and we need to decrease one after the SubnetPort creation is complete, which is done by the function ReleasePortInSubnet

Reply:
I see, thanks for the clarification.